### DISCUSSION TITLE
How to represent a boolean as two radio buttons?

### DISCUSSION THREAD

**Discussion by halljus** [edmundhung/conform#301] [Q&A] [answered]
Say you have a couple radio buttons that say "Yes" and "No" and you want these to map back to a boolean in the Zod schema. Would you recommend using the `conform.collection` utility somehow, or some other approach? Would representing a boolean as radio buttons require the use of Zod's `.preprocess`? Something else entirely?

Thanks.

**Comment by edmundhung** [comment, marked as answer]
You can use `.preprocess` or `.transform` like this:

```tsx
z.string().transform(value => value === 'yes')
// or
z.preprocess(value => value === 'yes', z.boolean())
```

Note: there seems to be a bug on zod v3.22 (latest version) with regards to `.preprocess`, so you might find it not working as expected.

**Reply by edmundhung** [reply]
The example above assumes you are setting value to yes on one of the radio buttons. You can change it to anything as you like.

**Reply by halljus** [reply]
Yeah, this is what I was thinking. Would you use `conform.collection` to render the options?

**Reply by edmundhung** [reply]
Yes. The collection helper is mainly designed for multiple checkboxes (as an array). But it will still be beneficial to use it for multiple radio buttons as it derives a unique id for each option you set.
